xarray/tests/test_netcdftimeindex.py

```diff
@@ -0,0 +1,349 @@
+import pytest
+
+import pandas as pd
+import xarray as xr
+
+from xarray.tests import assert_array_equal, assert_identical
+
+# Putting this at the module level for now, though technically we
+# don't need netCDF4 to test the string parser.
+pytest.importorskip('netCDF4')
+
+
+def date_dict(year=None, month=None, day=None,
+              hour=None, minute=None, second=None):
+    return dict(year=year, month=month, day=day, hour=hour,
+                minute=minute, second=second)
+
+
+@pytest.mark.parametrize(('string', 'expected'), [
+    ('1999', date_dict(year='1999')),
+    ('199901', date_dict(year='1999', month='01')),
+    ('1999-01', date_dict(year='1999', month='01')),
+    ('19990101', date_dict(year='1999', month='01', day='01')),
+    ('1999-01-01', date_dict(year='1999', month='01', day='01')),
+    ('19990101T12', date_dict(year='1999', month='01', day='01', hour='12')),
+    ('1999-01-01T12', date_dict(year='1999', month='01', day='01', hour='12')),
+    ('19990101T1234', date_dict(
+        year='1999', month='01', day='01', hour='12', minute='34')),
+    ('1999-01-01T12:34', date_dict(
+        year='1999', month='01', day='01', hour='12', minute='34')),
+    ('19990101T123456', date_dict(
+        year='1999', month='01', day='01', hour='12', minute='34',
+        second='56')),
+    ('1999-01-01T12:34:56', date_dict(
+        year='1999', month='01', day='01', hour='12', minute='34',
+        second='56')),
+    ('19990101T123456.78', date_dict(
+        year='1999', month='01', day='01', hour='12', minute='34',
+        second='56.78')),
+    ('1999-01-01T12:34:56.78', date_dict(
+        year='1999', month='01', day='01', hour='12', minute='34',
+        second='56.78'))
+], ids=['year', 'month', 'month-dash', 'day', 'day-dash', 'hour', 'hour-dash',
```

Comment:
This is a clever use of pytest but these two separate lists feels a little fragile. Maybe put these in a dict as a module level constant?